Sources/OfflineEntitlements/PurchasedSK2Product.swift

```diff
@@ -47,7 +47,7 @@ extension PurchasedSK2Product {
         self.entitlement = .init(
             expiresDate: expiration,
             productIdentifier: transaction.productID,
-            purchaseDate: Date(),
+            purchaseDate: transaction.purchaseDate,
```

Comment:
Note that this is an actual implementation change. However, using the transaction date might be more correct? This was introduced in https://github.com/RevenueCat/purchases-ios/pull/2358, but there are no details on this particular property. Let me know if this is actually not what we want.

Reply:
Seems this is only used for offline entitlements and conceptually it makes sense. Also checked on Android and seems this is the logic we use there so I think it's ok...

But I would like someone from @RevenueCat/catforms to double-check.

Reply:
Thanks Toni! Hi @RevenueCat/catforms, could you have a quick look? 🙏

Reply:
probably a remnant from the first drafts where we were just trying to make it work with whatever data was available, but I don't think there's any reason (or any wins) to make this `Date()`, transaction purchase date is more accurate

Reply:
That's good to know. Thanks for verifying!